File: l2-domain/src/main/java/com/vvygulyarniy/l2/domain/geo/Position.java
package com.vvygulyarniy.l2.domain.geo;

import lombok.Data;

/**
 * Created by Phoen-X on 16.02.2017.
 */
@Data
public class Position {
    private final Point point;
    private final int heading;

    public Position(int x, int y, int z) {
        this(x, y, z, 0);
    }

    public Position(int x, int y, int z, int heading) {
        this(new Point(x, y, z), heading);
    }

    public Position(final Point point, int heading) {
        this.point = point;
        this.heading = heading;
    }

    public double distanceTo(Position position) {
        return getPoint().distanceTo(position.getPoint());
    }
}

File: l2-domain/src/main/java/com/vvygulyarniy/l2/domain/geo/Position.kt
package com.vvygulyarniy.l2.domain.geo

data class Position(val point: Point, val heading: Int) {

    @JvmOverloads constructor(x: Int, y: Int, z: Int, heading: Int = 0) : this(Point(x, y, z), heading)

    fun distanceTo(position: Position): Double {
        return point.distanceTo(position.point)
    }
}
